File: domain/src/main/java/home/oleg/placenearme/domain/repositories/VenueRepository.java
package home.oleg.placenearme.domain.repositories;

import java.util.List;

import home.oleg.placenearme.domain.interactors.MapInteractor;
import home.oleg.placenearme.domain.models.Venue;
import io.reactivex.Observable;

public interface VenueRepository {

    Observable<List<Venue>> explore(MapInteractor.Parameters parameters);

    Observable<List<Venue>> search(String query);

}

File: core-domain/src/main/java/home/oleg/placesnearme/coredomain/repositories/VenueRepository.kt
package home.oleg.placesnearme.coredomain.repositories

import home.oleg.placesnearme.coredomain.models.Section
import home.oleg.placesnearme.coredomain.models.Venue
import io.reactivex.Single

interface VenueRepository {

    fun getRecommendedBySection(section: Section, requestParams: VenueRequestParams): Single<List<Venue>>

    fun search(query: String, requestParams: VenueRequestParams): Single<List<Venue>>

}
